ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh

```diff
@@ -0,0 +1,186 @@
+        fi
+    fi
+
+    return $result
+} >&2
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+else
+    echo "Unable to find kubeconfig under ${SHARED_DIR}!"
+    exit 1
+fi
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+ocp_version=$(getVersion)
+ocp_major_version=$( echo "${ocp_version}" | awk --field-separator=. '{print $1}' )
+ocp_minor_version=$( echo "${ocp_version}" | awk --field-separator=. '{print $2}' )
+echo "OCP Version: $ocp_version"
+
+# Mapping between optional capability and operators
+# Need to be updated when new operator marks as optional
+declare -A caps_operator
+caps_operator[baremetal]="baremetal"
+caps_operator[marketplace]="marketplace"
+caps_operator[openshift-samples]="openshift-samples"
+caps_operator[CSISnapshot]="csi-snapshot-controller"
+caps_operator[Console]="console"
+caps_operator[Insights]="insights"
+caps_operator[Storage]="storage"
+caps_operator[NodeTuning]="node-tuning"
+caps_operator[MachineAPI]="machine-api control-plane-machine-set cluster-autoscaler"
+
+v411="baremetal marketplace openshift-samples"
+# shellcheck disable=SC2034
+v412=" ${v411} Console Insights Storage CSISnapshot"
+v413=" ${v412} NodeTuning"
+v414=" ${v413} MachineAPI"
+latest_defined="v414"
+always_default="${!latest_defined}"
+
+# Determine vCurrent
+declare "v${ocp_major_version}${ocp_minor_version}"
+v_current_version="v${ocp_major_version}${ocp_minor_version}"
+
+if [[ ${!v_current_version:-} == "" ]]; then
+  echo "vCurrent: No default value for ${v_current_version}, use default value from ${latest_defined}: ${!latest_defined}"
+  vCurrent=${always_default}
+else
+  echo "vCurrent: Use exsting value from ${v_current_version}: ${!v_current_version}"
+  vCurrent=${!v_current_version}
+fi
+echo "vCurrent set: $vCurrent"
+
+enabled_capability_set=""
+case ${BASELINE_CAPABILITY_SET} in
+"None")
+  ;;
+"v4.11")
+  enabled_capability_set="${v411}"
+  ;;
+"v4.12")
+  enabled_capability_set="${v412}"
+  ;;
+"v4.13")
+  enabled_capability_set="${v413}"
+  ;;
+"v4.14")
+  enabled_capability_set="${v414}"
+  ;;
+"vCurrent")
+  enabled_capability_set="${vCurrent}"
+  ;;
+*)
+  enabled_capability_set="${always_default}"
+  ;;
+esac
+
+if [[ "${ADDITIONAL_ENABLED_CAPABILITIES}" != "" ]]; then
+    enabled_capability_set=$(echo ${enabled_capability_set} ${ADDITIONAL_ENABLED_CAPABILITIES} | xargs -n1 | sort -u | xargs)
+fi
+
+disabled_capability_set="${vCurrent}"
+for cap in $enabled_capability_set; do
+    disabled_capability_set=${disabled_capability_set/$cap}
+done
+
+echo "------cluster operators------"
+co_content=$(mktemp)
+oc get co | tee ${co_content}
+
+check_result=0
+# Enabled capabilities check
+echo "------check enabled capabilities-----"
+echo "enabled capability set: ${enabled_capability_set}"
+for cap in $enabled_capability_set; do
+    for op in ${caps_operator[$cap]}; do
+        if [[ ! `grep "${op}" ${co_content}` ]]; then
```

Comment:
@jinyunma It's better to match the operator's name more accurately:
e.g. we don't to match `operator-lifecycle-manager-catalog` if given `operator-lifecycle-manager`
```
operator-lifecycle-manager
operator-lifecycle-manager-catalog
operator-lifecycle-manager-packageserver
```


Note: there is a space(` `) after `${op}`
```suggestion
        if [[ ! `grep -e "^${op} " ${co_content}` ]]; then
```

Reply:
fixed